File: T4D4-IU-cs50-problems-2022-x-project/98306330-main/sentimental-mario-less/mario.py
from cs50 import get_int
while True:
    # pythonにはdo-while文が無いのでTrueでループさせ続けて最後に条件式で満たすモノをBreakで外に出してあげる事で擬似的にdo-while文を作る
    steps = get_int("Height:")
    if 1 < steps < 8:
        # 1より大きく8より小さい
        break
    elif steps == 1:
        # 1と同じ
        break
    elif steps == 8:
        # 8と同じ
        break
        # 3つの条件で1以上8以下の数字かどうかを判別
    print(f"please input figure between 1 and 8")
j = 1
for i in range(steps):
    print(" "*(steps-j), end='')
    print("#"*j)
    j += 1

File: T4D4-IU-cs50-problems-2022-x-project/98306330-main/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int steps = get_int("How many steps:");//段数を入力
    while (steps < 1 || steps > 8) //条件を満たさない場合に再度入力させる
    {
        steps = get_int("How many steps?:");
    }
    for (int i = 0; i < steps; i++)
    {
        for (int j = i + 1; j < steps; j++)
        {
            printf(" ");//空白を出力
        }
        for (int k = 8 - i; k < 9; k++)
        {
            printf("#");//#を出力
        }
        printf("\n");
    }
}